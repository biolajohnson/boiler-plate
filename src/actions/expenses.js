
import database from '../firebase/firebase'

//ADD_EXPENSE
export const addExpense = (expense) => ({
  type: "ADD_EXPENSE",
  expense
});
// async dispatch (database)
export const startAddExpense = (expenseData = {}) => {
  return (dispatch) => {
    const {
      description = '',
      amount = '',
      createdAt = 0,
      note = ''
    } = expenseData
    const expense = { description, amount, createdAt, note }
    return database.ref('expenses').push(expense).then((ref) => {
      dispatch(addExpense({
        id: ref.key,
        ...expense
      }))
    })
  }
}

//REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id,
});

//EDIT_EXPENSE
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates,
});
