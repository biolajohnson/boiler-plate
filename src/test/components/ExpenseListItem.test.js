import React from 'react'
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'
import ExpenseListItem from '../../components/ExpenseListItem'


test('should render individual enpense', () => { 
    const wrapper = shallow(<ExpenseListItem {...expenses[2]} />)
    expect(wrapper).toMatchSnapshot()
})