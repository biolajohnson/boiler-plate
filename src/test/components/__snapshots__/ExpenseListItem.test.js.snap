// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render individual enpense 1`] = `
<div>
  <Link
    to="/edit/3"
  >
    <h3>
      Cream Soda
    </h3>
  </Link>
  <p>
    150
     - 
    -345600000
  </p>
</div>
`;
